core/DataTable.php

```diff
@@ -1901,23 +1894,28 @@ protected function aggregateRowWithLabel(Row $row, $columnAggregationOps)
             throw new Exception($message);
         }
         $rowFound = $this->getRowFromLabel($labelToLookFor);
-        if ($rowFound === false) {
-            if ($labelToLookFor === self::LABEL_SUMMARY_ROW) {
-                $this->addSummaryRow($row);
+        $this->aggregateRow($rowFound, $row, $columnAggregationOps, $isSummaryRow = false);
+    }
+
+    private function aggregateRow($thisRow, Row $otherRow, $columnAggregationOps, $isSummaryRow)
+    {
+        if (empty($thisRow)) {
+            if ($isSummaryRow) {
+                $this->addSummaryRow($otherRow);
             } else {
-                $this->addRow($row);
+                $this->addRow($otherRow);
             }
         } else {
-            $rowFound->sumRow($row, $copyMeta = true, $columnAggregationOps);
+            $thisRow->sumRow($otherRow, $copyMeta = true, $columnAggregationOps);
```

Comment:
if $thisRow is not new and summary row will be aggregated based on the label or so, is it possible that we maybe would be aggregating a regular row with a summary row or the other way around and logic might not work 100% correct? Like would maybe need to call `setIsNotSummaryRow` or something and make sure it only aggregates summary rows together?

I'm thinking when it goes into `aggregateRowWithLabel()` and does `$rowFound = $this->getRowFromLabel($labelToLookFor);` then it might receive the summary row?

Reply:
> I'm thinking when it goes into aggregateRowWithLabel() and does $rowFound = $this->getRowFromLabel($labelToLookFor); then it might receive the summary row?

It might... we only do that for BC in getRowFromLabel(). Maybe we can introduce a getRowFromLabelWithoutSummaryRow() function or something?

Reply:
@tsteur updated w/ the smallest fix I could find.